src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -532,7 +533,11 @@ private void VisitNotEquals(BinaryExpression expression)
             }
 
             var memberInfo = GetMember(expression.Left);
-
+            if (memberInfo.Path == string.Empty)
+                throw new InvalidQueryException(
+                    "You cannot compare the whole document in Where closure. Please write a conditional statement using fields from the document.");
```

Comment:
```suggestion
                    "You cannot compare the whole document in Where closure. Please write a conditional statement using fields from the document. A query such as session.Query<User>().Where(u => u != null) is not meaningful, you are asserting that the document itself is not null, which it can never be. Did you intend to assert that a property isn't set to null? ");
```